context-js/src/lib.rs

```diff
@@ -134,3 +137,31 @@ pub async fn parse_meta_from_tarball(
         .await
         .map_err(|err| JsError::new(&err.to_string()))
 }
+
+#[wasm_bindgen()]
+pub async fn parse_internal_bin_from_tarball(
+    input: sys::ReadableStream,
+) -> Result<Vec<junit::bindings::BindingsReport>, JsError> {
+    let readable_stream = ReadableStream::from_raw(input);
+
+    // Many platforms do not support readable byte streams
+    // https://github.com/MattiasBuelens/wasm-streams/issues/19#issuecomment-1447294077
+    let async_read = match readable_stream.try_into_async_read() {
+        Ok(async_read) => Either::Left(async_read),
+        Err((_err, body)) => Either::Right(
+            body.into_stream()
+                .map_ok(|js_value| js_value.dyn_into::<Uint8Array>().unwrap_throw().to_vec())
+                .map_err(|_js_error| {
+                    std::io::Error::new(std::io::ErrorKind::Other, "failed to read")
+                })
+                .into_async_read(),
+        ),
+    };
+
+    let buf_reader = BufReaderAsync::new(async_read);
```

Comment:
Would it make sense to make a reusable function with this copied code?

Reply:
Still have not done this yet - will make another pass tomorrow